### PR TITLE
Enable Java parametric tests in CircleCI

### PR DESCRIPTION
**Pull request by bantonsson**
# What Does This Do

Enables Java parametric tests in CircleCI.

# Motivation

# Additional Notes

```
============================= test session starts ==============================
platform linux -- Python 3.9.16, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/circleci/project/system-tests/parametric, configfile: conftest.py
plugins: json-report-1.2.1, ddtrace-1.4.4, metadata-2.0.2, anyio-3.6.2
collected 111 items                                                            

test_headers_b3.py ....ssssss                                            [  9%]
test_headers_b3multi.py ........ssssssssssss                             [ 27%]
test_headers_datadog.py s....                                            [ 31%]
test_headers_none.py ..........                                          [ 40%]
test_headers_precedence.py ssss                                          [ 44%]
test_headers_tracecontext.py ssssssssssssssssssssssssssssssss            [ 72%]
test_headers_tracestate_dd.py sss                                        [ 75%]
test_library_tracestats.py .....ss..                                     [ 83%]
test_span_sampling.py .........s.....                                    [ 97%]
test_tracer.py .s.                                                       [100%]

================== 49 passed, 62 skipped in 232.81s (0:03:52) ==================
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -694,6 +695,40 @@ jobs:
...
+    resource_class: large
+    steps:
+      - setup_system_tests_repo
+
+      - run:
+          name: Install Python 3.9
+          command: |
+            pyenv install 3.9
+
+      - run:
```
Comment (smola) [inline .circleci/config.yml:711]: You'll need this:
```
      - restore_build_cache:
          cacheType: lib
```

at some point before this step.

**Review by smola — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (bantonsson) [reply .circleci/config.yml:714]: Why? Nothing in there builds locally. It's all containers in containers downloading the internet repeatedly.

Comment (bantonsson) [reply .circleci/config.yml:714]: Ah, you conflated the build cache and the workspace... I missed that in your PR. Is that really the best way?

Comment (bantonsson) [reply .circleci/config.yml:720]: I see why it has to be that way.

**Review by smola — APPROVED** [review]

**Pull request merged**